### PR TITLE
implement debug interface such as `store_info`, `cluster_info`,`dump_kv`

### PR DESCRIPTION
**Pull request by bufferflies**
### What is changed and how it works?

Issue Number: ref #14654

What's Changed:

```commit-message
debugger v2 impl `store_info`, `cluster_info`,`dump_kv`
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

Side effects

### Release note 

```release-note
debugger v2 impl `store_info`, `cluster_info`,`dump_kv`
```

### CODE REVIEW

File: src/server/debug2.rs [hunk 1]
```diff
@@ -442,6 +467,30 @@ impl<ER: RaftEngine> Debugger for DebuggerImplV2<ER> {
...
+    fn get_range_properties(&self, start: &[u8], end: &[u8]) -> Result<Vec<(String, String)>> {
+        let mut props = vec![];
+        let regions = find_region_states_by_key_range(&self.raft_engine, start, end);
+        for (region_id, start_key, end_key, region_state) in regions {
+            let mut tablet_cache =
+                get_tablet_cache(&self.tablet_reg, region_id, Some(region_state)).unwrap();
+            let talbet = tablet_cache.latest().unwrap();
+            let mut prop = dump_write_cf_properties(
+                talbet,
+                start_key.as_ref().map(|k| k.as_bytes()).unwrap_or(start),
```
Comment (SpadeA-Tang) [inline src/server/debug2.rs:480]: It seems this should be data_key format.

File: src/server/debug2.rs [hunk 2]
```diff
@@ -412,23 +381,79 @@ impl<ER: RaftEngine> Debugger for DebuggerImplV2<ER> {
...
+            .get_store_ident()
+            .transpose()
+            .unwrap()
+            .map_err(|e| Error::EngineTrait(e))
     }
 
-    fn get_region_properties(&self, _region_id: u64) -> Result<Vec<(String, String)>> {
-        unimplemented!()
+    fn get_region_properties(&self, region_id: u64) -> Result<Vec<(String, String)>> {
+        let region_state = self
```
Comment (SpadeA-Tang) [inline src/server/debug2.rs:424]: PeerState should be checked.

File: src/server/debug2.rs (hunk 1 above)
Comment (bufferflies) [reply src/server/debug2.rs]: good catch, fixed

File: src/server/debug2.rs (hunk 2 above)
Comment (bufferflies) [reply src/server/debug2.rs:424]: fixed

File: src/server/debug2.rs [hunk 3]
```diff
@@ -442,6 +470,30 @@ impl<ER: RaftEngine> Debugger for DebuggerImplV2<ER> {
...
+            let mut prop = dump_write_cf_properties(
+                talbet,
+                &keys::data_key(start_key.as_ref().map(|k| (k.as_bytes())).unwrap_or(start)),
+                &keys::data_key(end_key.as_ref().map(|k| k.as_bytes()).unwrap_or(end)),
+            )
+            .unwrap();
+            props.append(&mut prop);
+            let mut prop1 = dump_default_cf_properties(
+                talbet,
+                &keys::data_key(start_key.as_ref().map(|k| k.as_bytes()).unwrap_or(start)),
```
Comment (SpadeA-Tang) [inline src/server/debug2.rs:490]: If start_key is not None here, it is data_key already. It's better to add some tests for these cmds.

Comment (bufferflies) [reply src/server/debug2.rs]: good catch , fixed it and add unit test for it.

**Review by SpadeA-Tang — APPROVED** [review]

File: src/server/service/debug.rs [hunk 4]
```diff
@@ -21,6 +21,7 @@ fn error_to_status(e: Error) -> RpcStatus {
         Error::NotFound(msg) => (RpcStatusCode::NOT_FOUND, msg),
         Error::InvalidArgument(msg) => (RpcStatusCode::INVALID_ARGUMENT, msg),
         Error::Other(e) => (RpcStatusCode::UNKNOWN, format!("{:?}", e)),
+        Error::EngineTrait(e) => (RpcStatusCode::NOT_FOUND, format!("{:?}", e)),
```
Comment (overvenus) [inline src/server/service/debug.rs:24]:
```suggestion
        Error::EngineTrait(e) => (RpcStatusCode::UNKNOWN, format!("{:?}", e)),
```

I think it should be `UNKNOWN`, there are many other errors in `Error::EngineTrait`.

File: src/server/debug2.rs [hunk 5]
```diff
@@ -412,27 +381,81 @@ impl<ER: RaftEngine> Debugger for DebuggerImplV2<ER> {
...
 
-    fn get_region_properties(&self, _region_id: u64) -> Result<Vec<(String, String)>> {
-        unimplemented!()
-    }
+    fn get_region_properties(&self, region_id: u64) -> Result<Vec<(String, String)>> {
+        let region_state = self
+            .raft_engine
+            .get_region_state(region_id, u64::MAX)
+            .unwrap()
+            .unwrap();
```
Comment (pingyu) [inline src/server/debug2.rs:428]: Nit: suggest to return `Error::NotFound` on `None` for more friendly.

**Review by pingyu — COMMENTED** [review]
Rest LGTM.

File: src/server/debug2.rs (hunk 5 above)
Comment (bufferflies) [reply src/server/debug2.rs]: fixed

File: cmd/tikv-ctl/src/executor.rs [hunk 6]
```diff
@@ -1236,40 +1236,57 @@ impl<ER: RaftEngine> DebugExecutor for DebuggerImplV2<ER> {
...
+    fn drop_unapplied_raftlog(&self, _region_ids: Option<Vec<u64>>) {}
 
-    fn recreate_region(&self, _sec_mgr: Arc<SecurityManager>, _pd_cfg: &PdConfig, _region_id: u64) {
+    fn recreate_region(&self, _mgr: Arc<SecurityManager>, _pd_cfg: &PdConfig, _region_id: u64) {
         unimplemented!()
     }
 
     fn dump_metrics(&self, _tags: Vec<&str>) {
-        unimplemented!()
+        unimplemented!("only available for online mode");
```
[the remark this answers is not in the document]
Comment (bufferflies) [reply cmd/tikv-ctl/src/executor.rs]: fixed

**Review by pingyu — APPROVED** [review]
LGTM~

**Pull request merged**